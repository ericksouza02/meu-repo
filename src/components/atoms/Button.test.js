import React from "react";
import {screen,fireEvent } from '@testing-library/react';
import { render } from "@testing-library/react";
import ThemeProvider from '../../styles/ThemeProvider';
import '@testing-library/jest-dom';
import Button from "./Button"

test('element children in the Button', () => {
    const { getByText } = render(
    <ThemeProvider>
        <Button>Clique Aqui</Button>
    </ThemeProvider>
)
    expect(screen.getByText('Clique Aqui')).toBeInTheDocument()
});

test('event click on Button', () => {

    const handleCLick = jest.fn()

    const {getByRole} = render(
    <ThemeProvider>
        <Button onClick={handleCLick}/>
    </ThemeProvider>
    )

    fireEvent.click(screen.getByRole('button')) // paramentros fireEvent.click(elemento, evento)

    expect(handleCLick).toBeCalled()
});

// Jeito de fazer com que o teste nÃ£o quebre
// Object.values(ButtonColors).map(item=>[item])

test.each([
   ['primary', 'outlined'],
   ['primary', 'link'],
   ['danger', 'outlined'],
   ['danger', 'link'],
   ['default', 'outlined'],
   ['default', 'link']
])('renders with %i color', (color, variant) => {

    const { asFragment } = render(
    <ThemeProvider>
        <Button color={color} variant={variant}/>
    </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
});