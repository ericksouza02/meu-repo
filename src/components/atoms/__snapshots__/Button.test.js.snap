// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with NaN color 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnyFh sc-gtsqUy kOmast cXYVZE"
    color="primary"
    type="button"
    variant="outlined"
  />
</DocumentFragment>
`;

exports[`renders with NaN color 2`] = `
<DocumentFragment>
  <button
    class="sc-bdnyFh sc-dlniIP kOmast gBbkIs"
    color="primary"
    type="button"
    variant="link"
  />
</DocumentFragment>
`;

exports[`renders with NaN color 3`] = `
<DocumentFragment>
  <button
    class="sc-bdnyFh sc-gtsqUy idYrkJ bZQxaM"
    color="danger"
    type="button"
    variant="outlined"
  />
</DocumentFragment>
`;

exports[`renders with NaN color 4`] = `
<DocumentFragment>
  <button
    class="sc-bdnyFh sc-dlniIP idYrkJ fZBeuw"
    color="danger"
    type="button"
    variant="link"
  />
</DocumentFragment>
`;

exports[`renders with NaN color 5`] = `
<DocumentFragment>
  <button
    class="sc-bdnyFh sc-gtsqUy ehimHh gkqJIw"
    color="default"
    type="button"
    variant="outlined"
  />
</DocumentFragment>
`;

exports[`renders with NaN color 6`] = `
<DocumentFragment>
  <button
    class="sc-bdnyFh sc-dlniIP ehimHh oVjZZ"
    color="default"
    type="button"
    variant="link"
  />
</DocumentFragment>
`;
